server/routerlicious/packages/services/src/mongodb.ts

```diff
@@ -7,103 +7,174 @@ import { assert } from "console";
 import * as core from "@fluidframework/server-services-core";
 import { AggregationCursor, Collection, MongoClient, MongoClientOptions } from "mongodb";
 import { Lumberjack } from "@fluidframework/server-services-telemetry";
+import { requestWithRetry } from "@fluidframework/server-services-core";
+import { MongoErrorRetryAnalyzer } from "./mongoExceptionRetryRules";
 
 const MaxFetchSize = 2000;
 
-export class MongoCollection<T> implements core.ICollection<T> {
-    constructor(private readonly collection: Collection<T>) {
+export class MongoCollection<T> implements core.ICollection<T>, core.IRetryAble {
+    public readonly retryEnabled: boolean;
+    constructor(
+        private readonly collection: Collection<T>,
+        private readonly clientSideRetry?: boolean,
+    ) {
+        assert(collection, "collection must be defined");
```

Comment:
Static type checking should be enough to prevent this, since `undefined` is not allowed in the param types. Why do we need to assert?

Reply:
I don't know, I suspect it my IDE auto completed added this line...

Reply:
deleted in b0e2d37